### ISSUE TITLE
TypedDict unpacking (**data) incorrectly flagged as type error when parameters match TypedDict fields

### ISSUE THREAD

**Issue by FabianVegaA** [astral-sh/ty#663]
### Summary

When unpacking a TypedDict instance using the `**` operator into a function whose parameters exactly match the TypedDict's field types, the type checker incorrectly reports a type error. This happens despite the structural compatibility between the TypedDict definition and the function signature.

Example:
```python
from typing import TypedDict

class Foo(TypedDict):
    bar: int
    baz: str
    qux: float
    quux: tuple[int]

def do_with_foo(
    bar: int,
    baz: str,
    qux: float,
    quux: tuple[int],
) -> int:
    raise NotImplemented

data = Foo(bar=1, baz="1", qux=1.1, quux=(1,))
result = do_with_foo(**data) # Type checker incorrectly flags this line with error
```

The error marked is **"No arguments provided for required parameters `bar`, `baz`, `qux`, `quux` of function `do_with_foo` (ty missing-argument)"**

I'm using Zed integration. This is my configuration:
```json
{
"languages": {
    "Python": {
      "language_servers": ["ty"]
    }
  },
  "lsp": {
    "ty": {
      "binary": {
        "path": "/path/to/bin/ty",
        "arguments": ["server"]
      }
    }
}
```

### Version

ty 0.0.1-alpha.10 (15bae14c2 2025-06-13)

**Comment by abhijeetbodas2001** [comment]
Related: https://github.com/astral-sh/ty/issues/154

**Comment by carljm** [comment]
Also related: https://github.com/astral-sh/ty/issues/247

Thanks for the report, but since this is a combination of two not-yet-supported features (TypedDict and splatted calls), I'm going to mark it as duplicate; if it still doesn't work after the above two issues are closed, then there may be a bug in one of the implementations of those features.
